pkg/controller/networkpolicy/networkpolicy_controller_test.go

```diff
@@ -3139,6 +3143,242 @@ func TestSyncInternalNetworkPolicyConcurrently(t *testing.T) {
+				{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "groupA"},
+					Spec:       v1alpha3.GroupSpec{PodSelector: &selectorA},
+				},
+			},
+			inputPolicy: &crdv1alpha1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "anpA", UID: "uidA"},
+				Spec: crdv1alpha1.NetworkPolicySpec{
+					AppliedTo: []crdv1alpha1.NetworkPolicyPeer{
+						{Group: "groupA"},
+					},
+					Priority: p10,
+					Ingress: []crdv1alpha1.Rule{
+						{
+							From:   []crdv1alpha1.NetworkPolicyPeer{{PodSelector: &selectorB}},
+							Action: &allowAction,
+						},
+					},
+				},
+			},
+			expectedPolicy: &antreatypes.NetworkPolicy{
+				UID:      "uidA",
+				Name:     "uidA",
+				SpanMeta: antreatypes.SpanMeta{NodeNames: sets.NewString("nodeA")},
+				SourceRef: &controlplane.NetworkPolicyReference{
+					Type:      controlplane.AntreaNetworkPolicy,
+					Namespace: "nsA",
+					Name:      "anpA",
+					UID:       "uidA",
+				},
+				Priority:     &p10,
+				TierPriority: &DefaultTierPriority,
+				Rules: []controlplane.NetworkPolicyRule{
+					{
+						Direction: controlplane.DirectionIn,
+						From:      controlplane.NetworkPolicyPeer{AddressGroups: []string{selectorBGroup}},
+						Action:    &allowAction,
+					},
+				},
+				AppliedToGroups: []string{"nsA/groupA"},
+			},
+		},
+		{
+			name: "anp with valid parent group",
+			groups: []*v1alpha3.Group{
+				{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "parentGroup"},
+					Spec:       v1alpha3.GroupSpec{ChildGroups: []v1alpha3.ClusterGroupReference{"groupA"}},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "groupA"},
+					Spec:       v1alpha3.GroupSpec{PodSelector: &selectorA},
+				},
+			},
+			inputPolicy: &crdv1alpha1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "anpA", UID: "uidA"},
+				Spec: crdv1alpha1.NetworkPolicySpec{
+					AppliedTo: []crdv1alpha1.NetworkPolicyPeer{
+						{Group: "parentGroup"},
+					},
+					Priority: p10,
+					Ingress: []crdv1alpha1.Rule{
+						{
+							From:   []crdv1alpha1.NetworkPolicyPeer{{PodSelector: &selectorB}},
+							Action: &allowAction,
+						},
+					},
+				},
+			},
+			expectedPolicy: &antreatypes.NetworkPolicy{
+				UID:      "uidA",
+				Name:     "uidA",
+				SpanMeta: antreatypes.SpanMeta{NodeNames: sets.NewString("nodeA")},
+				SourceRef: &controlplane.NetworkPolicyReference{
+					Type:      controlplane.AntreaNetworkPolicy,
+					Namespace: "nsA",
+					Name:      "anpA",
+					UID:       "uidA",
+				},
+				Priority:     &p10,
+				TierPriority: &DefaultTierPriority,
+				Rules: []controlplane.NetworkPolicyRule{
+					{
+						Direction: controlplane.DirectionIn,
+						From:      controlplane.NetworkPolicyPeer{AddressGroups: []string{selectorBGroup}},
+						Action:    &allowAction,
+					},
+				},
+				AppliedToGroups: []string{"nsA/parentGroup"},
+			},
+		},
+		{
+			name: "anp with invalid group",
```

Comment:
maybe a more verbose name would be useful to understand why the group is invalid in this case: "anp with invalid group selecting pods in multiple Namespaces"

same for the test case below

Reply:
updated, thanks